src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -660,6 +661,8 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
                 return false;
             }
             ((CreateIndexRequest) request).index(newIndices.length!=1?null:newIndices[0]);
+        } else if (request instanceof CreateDataStreamAction.Request) {
+            provider.provide(((CreateDataStreamAction.Request) request).indices(), request, true);
```

Comment:
Should it be similar to `CreateIndexRequest` handling?

Reply:
Checked with @sandeshkr419: `CreateDataStreamAction.Request` unlike `CreateIndexRequest` does not support request index replace, so it should be handled similar to `BulkShardRequest`, not `CreateIndexRequest`. 

@sandeshkr419 even though the last parameter won't be used, can you change it to `false` and add comment explaining that the request is handled similar to `BulkShardRequest`.

Reply:
Referring to BulkShardRequest can add to confusion here. 

To explain it better, the only difference between handling of `CreateIndexRequest` & `CreateDataStreamAction.Request` is that the latter does not supports index/data-stream replace functionality for now and therefore the analogous call to replace index (like in `CreateIndexRequest`) is not made in `CreateDataStreamAction.Request`.

I have changed the last parameter to false now.